pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,443 @@
+	for key, value := range currentLv {
+		verbosityVal[key] = value
+	}
+
+	// create a message to show verbosity for the specified component
+	lines := createOutputLines(verbosityVal, options)
+
+	return lines
+}
+
+func addPatch(patchData *[]patch.PatchOperation, currentLv map[string]uint) {
+	*patchData = append(*patchData, patch.PatchOperation{
+		Op:    patchOperation,
+		Path:  patchPath,
+		Value: currentLv,
+	})
+}
+
+func setVerbosity(currentLv, options map[string]uint, patchData *[]patch.PatchOperation) {
+	// update currentLv based on the user-specified verbosity for all components
+	if verbosity, exist := options["all"]; exist {
+		for component := VirtAPI; component < All; component++ {
+			JSONName := getJSONNameByVirtComponent(component)
+			currentLv[JSONName] = verbosity
+		}
+	}
+
+	// update currentLv based on the user-specified verbosity for each component
+	for componentName, verbosity := range options {
+		if componentName == "all" {
+			continue
+		}
+		JSONName := getJSONNameByComponentName(componentName)
+		currentLv[JSONName] = verbosity
+	}
+
+	// in case of just reset (no set operation after the reset), don't need to add another patch
+	if len(currentLv) != 0 {
+		addPatch(patchData, currentLv)
+	}
+}
+
+func createPatch(currentLv, options map[string]uint) ([]byte, error) {
+	patchData := []patch.PatchOperation{}
+
+	// reset only if verbosity exists, otherwise do nothing
+	if isReset && len(currentLv) != 0 {
+		// add an empty object
+		currentLv = map[string]uint{}
+		addPatch(&patchData, currentLv)
+	}
+
+	setVerbosity(currentLv, options, &patchData)
+
+	return json.Marshal(patchData)
+}
+
+func findOperation(cmd *cobra.Command, options map[string]uint) (operation, error) {
+	isShow, isSet := false, false
+
+	for component := VirtAPI; component <= All; component++ {
+		componentName := GetComponentNameByVirtComponent(component)
+
+		// check if the flag for the component is specified
+		if !cmd.Flags().Changed(componentName) {
+			continue // do nothing for the component
+		}
+
+		// if flag is specified, it means either set or show
+		// if the value = noArg, it means show
+		// if the value != noArg, it means set
+		isShow = isShow || verbosities[component] == noArg
+		isSet = isSet || verbosities[component] != noArg
+
+		// check whether the verbosity is in the range
+		// allow noArg (=10)
+		// because user can see the value 10 as the default in the help like this:
+		// 		--all uint[=10] show/set all component log verbosity (0-9) (default 11)
+		// --all=10, which equals -all (show operation)
+		if verbosities[component] != noArg && verbosities[component] > maxVerbosity {
+			return nop, fmt.Errorf("%s: log verbosity must be %d-%d", componentName, minVerbosity, maxVerbosity)
+		}
+
+		// add a componentName and its verbosity to the map
+		// in case of show: verbosity = noArg
+		// in case of set: verbosity = 0-9
+		// in case of reset: no information added to the map
```

Comment:
Is this comment necessary?

Reply:
In the case of Reset, I think it's not clear, so I'll leave the line of Reset, and remove other lines.